### DISCUSSION TITLE
Error retrieving taxonomy file that does appear to exist

### DISCUSSION THREAD

**Discussion by ahoward-cf** [manusimidt/py-xbrl#93] [Q&A] [answered]
I am encountering the following error when trying to parse an iXBRL file saved locally. With other files, the parser is able to download all the relevant taxonomies, but this file (and many others like it) are failing when trying to retrieve certain taxonomies. The error appears to indicate that such a taxonomy does not exist, but following the link indicates that there is something present. My knowledge of XBRL is limited and I am not sure what is causing this error. 

My code block is as follows:

`logging.basicConfig(level=logging.INFO)

cache: HttpCache = HttpCache('./cache')
parser = XbrlParser(cache)

schema_url = "./07804038_aa_2021-08-27.xhtml"
inst: XbrlInstance = parser.parse_instance(schema_url)`

The taxonomy is linked at [this address](https://xbrl.frc.org.uk/FRS-102/2019-01-01/FRS-102-2019-01-01.xsd) which does appear to exist. 

Yet I get the following error.

![image](https://user-images.githubusercontent.com/29404906/198009841-440bc83a-15e3-4f0b-bfd9-0d41c2ace521.png)

**Comment by manusimidt** [comment]
The error code 404 implies that the file could not be found on the server. However, I was also able to retrieve the file using the url you provided.

Have you tried setting the HTTP header? https://py-xbrl.readthedocs.io/en/latest/usage.html#online
Maybe the server blocks your request because of some bot-limitations. 

If you want you can send me the xbrl file via email and i will try to parse it myself.

**Reply by manusimidt** [reply]
Thank you for supplying the file.
This is actually a very stupid bug in `py-xbrl`. The filer has for some reason but a space after the url which gets converted to "%20" by the requests libary 😅
I will quickly implement a fix. 

![image](https://user-images.githubusercontent.com/29599104/198370718-460e9ea8-1531-4bf3-96eb-796659f05e35.png)

Please mark this and the other discussion as answered if it is answered for you :)

**Comment by manusimidt** [comment, marked as answer]
@ahoward-cf Unfortunately the parser is still not able to parse the instance document you provided. 
The issue is that the filer uses the prefix 'dpl-frs' but does not specify a namespace for it. (the `xmlns:dpl-frs="..."` is missing). 

This is not only invalid XBRL but also invalid with regards to the [XML Specification](https://www.w3schools.com/xml/xml_namespaces.asp), as each prefix must have a namespace.
So unfortunately there is no way to parse this particular filing. Filings from bigger companies and/or from trustworthy sources (as EDGAR or UK Company House) should not be syntactically incorrect.

**Reply by ahoward-cf** [reply]
I was just about to message you to see if you had been able to parse it in the end, but then I found this. Clearly I missed it the first time. 

That is... annoying. Not with your package, with the filer. Interestingly enough, this XBRL is directly from the UK Companies House website. Perhaps I could suggest they use your package as a verification tool!
